src/generic/listctrl.cpp

```diff
@@ -2197,6 +2205,57 @@ void wxListMainWindow::HighlightAll( bool on )
     }
 }
 
+void wxListMainWindow::HighlightOnly( size_t line, size_t oldLine )
+{
+    const unsigned selCount = GetSelectedItemCount();
+
+    if ( selCount == 1 && IsHighlighted(line) )
+    {
+        return; // Nothing changed.
+    }
+
+    if ( oldLine != (size_t)-1 )
+    {
+        IsHighlighted(oldLine) ? ReverseHighlight(oldLine)
+                               : RefreshLine(oldLine); // refresh the old focus to remove it
+    }
+
+    if ( selCount > 1 ) // multiple-selection only
+    {
+        // Deselecting many items at once will generate wxEVT_XXX_DESELECTED event
+        // for each one of them. although this may be inefficient if the number of
+        // deselected items is too much, we keep doing this (for non-virtual list
+        // controls) for backward compatibility concerns. For virtual listctrl (in
+        // multi-selection mode), wxMSW sends only a notification to indicate that
+        // something has been deselected. Notice that to be fully compatible with
+        // wxMSW behaviour, _line_ shouldn't be deselected if it was selected.
+
+        const SendEvent sendEvent = IsVirtual() ? SendEvent_None : SendEvent_Normal;
+
+        size_t lineFrom = 0,
+               lineTo   = GetItemCount() - 1;
+
+        if ( line > lineFrom && line < lineTo )
+        {
+            HighlightLines(lineFrom, line - 1, false, sendEvent);
+            HighlightLines(line + 1, lineTo, false, sendEvent);
+        }
+        else // _line_ is equal to lineFrom or lineTo
+        {
+            line == lineFrom ? ++lineFrom : --lineTo;
+            HighlightLines(lineFrom, lineTo, false, sendEvent);
+        }
+
+        if ( IsVirtual() )
```

Comment:
```suggestion
        // If we didn't send the event for individual items above, send it for all of them now.
        if ( sendEvent == SendEvent_None )
```
Just to make it slightly more clear.